workflow/executor/pns/pns.go

```diff
@@ -331,12 +305,15 @@ func (p *PNSExecutor) secureRootFiles() error {
 				return err
 			}
 
-			// the main container may have switched (e.g. gone from busybox to the user's container)
-			if prevInfo, ok := p.pidFileHandles[pid]; ok {
-				_ = prevInfo.Close()
+			if p.pidFileHandles[pid] != fs {
+				// the main container may have switched (e.g. gone from busybox to the user's container)
+				if prevInfo, ok := p.pidFileHandles[pid]; ok {
+					_ = prevInfo.Close()
+				}
+				p.pidFileHandles[pid] = fs
+				log.Infof("ALEX secured root for pid %d root: %s (%q)", pid, proc.Executable(), fs.Name())
```

Comment:
Just curious. Is this part `ALEX ..` for debugging puporse?

Reply:
ops...